### PR TITLE
[Ryan Aidan] iP

### PR DESCRIPTION
**Pull request by aidanaden**
DukePro2 frees your mind of having to remember things you need to do. It's

* text-based
* easy to learn
* ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 👍

And it is **FREE**

Features:
* [x]  Managing tasks
* [x]  Managing deadlines
* [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

### CODE REVIEW

File: src/main/java/duke/Command.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    MARK,
+    UNMARK,
+    DEADLINE,
+    EVENT,
+    TODO,
+    DELETE,
+    FIND,
+    BYE,
+    ERROR
+}
```
Comment (atmh) [inline src/main/java/duke/Command.java]: I like that you have included enum 👍.

File: src/main/java/duke/Parser.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                return Command.EVENT;
+            } else if (commandString.equals("todo")) {
+                return Command.TODO;
+            } else if (commandString.equals("delete")) {
+                return Command.DELETE;
+            } else if (commandString.equals("find")) {
+                return Command.FIND;
+            } else {
+                return Command.ERROR;
+            }
```
Comment (atmh) [inline src/main/java/duke/Parser.java]: Perhaps a switch statement would be more succinct. E.g. 
```java
switch (commandString) {
case "bye":
    ...
    break;
case "list":
    ...
    break;
default:
    ...
}
```

File: src/main/java/duke/DeleteEmptyException.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package duke;
+
+/**
+ * Exception when user attempts to delete a task
+ * without providing the index of the task
+ */
+public class DeleteEmptyException extends DukeException {
```
Comment (atmh) [inline src/main/java/duke/DeleteEmptyException.java:7]: I like that you have created classes for specific exceptions

File: src/main/java/duke/Duke.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+        try {
+            tasks =  new TaskList(storage.loadListFromDisk());
+        } catch (FileNotFoundException err) {
+            ui.displayLoadError();
+            tasks = new TaskList();
+        }
+    }
+
+    public String[] getEventTimings(String text) {
```
Comment (atmh) [inline src/main/java/duke/Duke.java]: Just a little more Javadocs to go 👍

File: src/main/java/duke/Storage.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
+package duke;
```
Comment (atmh) [inline src/main/java/duke/Storage.java:1]: Although it's not required, I feel that dividing the classes into multiple packages would be cleaner. 
E.g. `duke.storage`, `duke.task`, `duke.ui`, `duke.command`

File: src/main/java/duke/Task.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+    /**
+     * Name of task
+     */
+    protected String description;
+
+    /**
+     * Completion status of task
+     */
+    protected boolean isDone;
```
Comment (atmh) [inline src/main/java/duke/Task.java:19]: Overall, the code follows the naming guidelines very closely 👍.

**Review by atmh — COMMENTED** [review]
Looks good 😄.
The code abides to the code quality guidelines 👍, just some minor nits to clean up.

File: src/main/java/duke/Parser.java (hunk 2 above)
Comment (aidanaden) [reply src/main/java/duke/Parser.java]: but switch statements look more complicated to me :(

File: src/main/java/duke/DukeException.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+ * <p>The class {@code duke.DukeException} and any subclasses that are not
+ * also subclasses of {@link RuntimeException} are <em>checked
+ * exceptions</em>.  Checked exceptions need to be declared in a
+ * method or constructor's {@code throws} clause if they can be thrown
+ * by the execution of the method or constructor and propagate outside
+ * the method or constructor boundary.
+ */
+public class DukeException extends Exception {
+    public DukeException(String err) {
+        super("\n############################################################\n ☹ OOPS!!! "
```
Comment (tanchinkiat99) [inline src/main/java/duke/DukeException.java:18]: Will it be better to have a more comprehensive error message?

File: src/main/java/duke/Event.java [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        return getStartTime().format(DateTimeFormatter.ofPattern(formatString));
+    }
+
+    /**
+     * Return start datetime as a string in the format dd/MM/yyyy HHmm
+     *
+     * @return Start datetime as a string using the default format
+     */
+    public String getStartTimeString() {
+        return getStartTimeString("dd/MM/yyyy HHmm");
```
Comment (tanchinkiat99) [inline src/main/java/duke/Event.java]: Will it be better to combine this with the above function?

File: src/main/java/duke/Parser.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+     * @param input User input text
+     * @return Command type
+     */
+    public static Command parse(String input) {
+        String[] inputSplit = input.split(" ");
+
+        if (inputSplit.length > 0) {
+            String commandString = inputSplit[0];
+
+            if (commandString.equals("bye")) {
```
Comment (tanchinkiat99) [inline src/main/java/duke/Parser.java:30]: Is it more suitable to use enumeration and switch function here?

File: src/main/java/duke/Storage.java [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+            Scanner f = new Scanner(defaultFile);
+            ArrayList<Task> data = new ArrayList<>();
+
+            while (f.hasNext()) {
+                String[] taskLineSplit = f.nextLine().split(",");
+                Boolean isCompleted = Integer.parseInt(taskLineSplit[1]) == 1 ? true : false;
+                String taskName = taskLineSplit[2];
+                Task storedTask;
+
+                if (taskLineSplit[0].equals("T")) {
```
Comment (tanchinkiat99) [inline src/main/java/duke/Storage.java]: I like how you used comments to show the functionality of each statement. 😄

**Review by tanchinkiat99 — COMMENTED** [review]
Looks good. Just a few possible improvements here and there to help make your code more organised.

**Pull request closed without merging**